projects/helper/rpcFallback.js

```diff
@@ -0,0 +1,86 @@
+const { getEnv } = require('./env')
+const sdk = require('@defillama/sdk')
+const axios = require('axios')
+
+const uniq  = (arr) => [...new Set(arr.filter(Boolean))]
+const cache = Object.create(null)
+
+function buildRpcAggregateError (chain, failures) {
+  const aggErr = new AggregateError(
+    failures.map(f => f.err),
+    `[${chain}] all RPC endpoints failed`
+  )
+
+  aggErr.chain = chain
+  aggErr.failures = failures.map(({ url, err }) => ({ url, message : err.message, code : err.code, stack : err.stack }))
+
+  return aggErr
+}
+
+function buildRpcList(chain) {
+  if (cache[chain]) return cache[chain]
+
+  const list = []
+
+  const agg = getEnv('RPC_AGGREGATOR_URL')
+  
+  if (agg) {
+    const aggUrl = agg.includes('{chain}')
+      ? agg.replace('{chain}', chain)
+      : `${agg}/${chain}`
+    list.push(aggUrl)
+  }
+
+  const envKey = `${chain.toUpperCase()}_RPC`
+  const direct = getEnv(envKey)
+  
+  if (direct) {
+    const directUrls = direct.split(',')
+    list.push(...directUrls)
+  }
+
+  cache[chain] = uniq(list)
+  return cache[chain]
+}
+
+function createAxiosInstance(url) {
```

Comment:
instead of this, we maintain the list in line 23 as axiosObject array? 

that way, we avoid the url check & interceptors here, and instead do 

    if (agg) {
      const aggUrl = agg.includes('{chain}') ? agg.replace('{chain}', chain) : `${agg}/${chain}`
      const aggInstance = axios.create({
        baseURL: aggUrl,
        query_parameters: {
          source: 'tvl_adapter'
        },
      })
      list.push(aggInstance)
    }